proxy/src/set_public_ip_and_start.sh

```diff
@@ -9,11 +9,20 @@
 # with the the real public ip. There's an order of priority here which is
 # 1. Environment variable
 # 2. AWS EC2 Metadata endpoint
-#
+# 3. Third-party sources
 # If all fails, we'll just not set the destination IP address
 
 CONFIG_FILE="/usr/local/etc/haproxy/haproxy.cfg"
 
+## Custom function to use as curl wrapper
+# --silent: to reduce the nois eof response
+# --show-error: to show errors in the response
+# --fail: to fail on non-200 responses
+# --ipv4: to force ipv4 resolution
+function fetch() {
+  curl --silent --show-error --fail --ipv4 "$@"
```

Comment:
nit: can we put the `--max-time 2` arg into here and remove it from the callsites?

Reply:
yes sure, will update it now.